src/handlers/artist.rs

```diff
@@ -0,0 +1,206 @@
+    if let Some(artist) = &mut app.artist {
+        match artist.artist_selected_block {
+            ArtistBlock::TopTracks => {
+                let next_index = common_key_events::on_up_press_handler(
+                    &artist.top_tracks,
+                    Some(artist.selected_top_track_index),
+                );
+                artist.selected_top_track_index = next_index;
+            }
+            ArtistBlock::Albums => {
+                let next_index = common_key_events::on_up_press_handler(
+                    &artist.albums.items,
+                    Some(artist.selected_album_index),
+                );
+                artist.selected_album_index = next_index;
+            }
+            ArtistBlock::RelatedArtists => {
+                let next_index = common_key_events::on_up_press_handler(
+                    &artist.related_artists,
+                    Some(artist.selected_related_artist_index),
+                );
+                artist.selected_related_artist_index = next_index;
+            }
+            ArtistBlock::Empty => {}
+        }
+    }
+}
+
+fn handle_up_press_on_hovered_block(app: &mut App) {
+    if let Some(artist) = &mut app.artist {
+        match artist.artist_hovered_block {
+            ArtistBlock::TopTracks => {
+                artist.artist_hovered_block = ArtistBlock::RelatedArtists;
+            }
+            ArtistBlock::Albums => {
+                artist.artist_hovered_block = ArtistBlock::TopTracks;
+            }
+            ArtistBlock::RelatedArtists => {
+                artist.artist_hovered_block = ArtistBlock::Albums;
+            }
+            ArtistBlock::Empty => {}
+        }
+    }
+}
+
+fn handle_enter_event_on_selected_block(app: &mut App) {
+    if let Some(artist) = &mut app.artist.clone() {
+        match artist.artist_selected_block {
+            ArtistBlock::TopTracks => {
+                let selected_index = artist.selected_top_track_index;
+                let selected_track = &artist.top_tracks[selected_index];
+                app.start_playback(None, Some(vec![selected_track.uri.to_owned()]), Some(0));
+            }
+            ArtistBlock::Albums => {
+                if let Some(selected_album) = artist
+                    .albums
+                    .items
+                    .get(artist.selected_album_index)
+                    .cloned()
+                {
+                    app.track_table.context = Some(TrackTableContext::AlbumSearch);
+                    app.get_album_tracks(selected_album);
+                }
+            }
+            ArtistBlock::RelatedArtists => {
+                let selected_index = artist.selected_related_artist_index;
+                let artist_id = &artist.related_artists[selected_index].id;
+                let artist_name = &artist.related_artists[selected_index].name;
+                app.get_artist(&artist_id, &artist_name);
+                // app.get_artist_related_artists(&artist_id);
```

Comment:
Are you able to delete this commented code?

Reply:
Sorry, I had forgotten to delete commented codes. I have deleted them.